### PR TITLE
ADD api_key authentication to bugzilla.py  

### PR DESCRIPTION
**Pull request by K-Gazdikova**
- Modified  authentication for bugzilla.py  to include api_key method  with no separate classes .

### CODE REVIEW

File: ticketutil/bugzilla.py [hunk 1]
```diff
@@ -1,11 +1,24 @@
 import logging
+import os
 
 import requests
 
+import base64
```
Comment (dmranck) [inline ticketutil/bugzilla.py]: This isn't used anywhere. You can remove for this PR (I think you might use this in a future PR for add_attachment()?)

File: ticketutil/bugzilla.py [hunk 2]
```diff
@@ -1,11 +1,24 @@
 import logging
+import os
```
Comment (dmranck) [inline ticketutil/bugzilla.py]: You can remove this because we'll be removing lines 12-21 below.

File: ticketutil/bugzilla.py [hunk 3]
```diff
@@ -1,11 +1,24 @@
...
 
 import requests
 
+import base64
+
 from . import ticket
 
 __author__ = 'dranck, rnester, kshirsal'
 
+# Disable warnings for requests because we aren't doing certificate verification
```
Comment (dmranck) [inline ticketutil/bugzilla.py]: You can remove lines 12-21. This is handled in ticket.py now, which all the other tool-specific files import.

File: ticketutil/bugzilla.py [hunk 4]
```diff
@@ -138,10 +180,7 @@ def _create_ticket_request(self, params):
         """
         # Attempt to create ticket.
         try:
-            if self.token:
-                params['token'] = self.token
-
-            r = self.s.post(self.rest_url, json=params)
+            r = self.s.post(self.rest_url, json=params, verify=False)
```
Comment (dmranck) [inline ticketutil/bugzilla.py]: remove verify=False, because this is handled on the session-level on line 80.

File: ticketutil/bugzilla.py [hunk 5]
```diff
@@ -184,14 +223,11 @@ def edit(self, **kwargs):
...
         kwargs = _prepare_ticket_fields(kwargs)
-
         params = kwargs
-        if self.token:
-            params['token'] = self.token
 
         # Attempt to edit ticket.
         try:
-            r = self.s.put("{0}/{1}".format(self.rest_url, self.ticket_id), json=params)
+            r = self.s.put("{0}/{1}".format(self.rest_url, self.ticket_id), json=params, verify=False)
```
Comment (dmranck) [inline ticketutil/bugzilla.py]: remove verify=False, because this is handled on the session-level on line 80.

File: ticketutil/bugzilla.py [hunk 6]
```diff
@@ -202,11 +238,12 @@ def edit(self, **kwargs):
                 return
             logging.debug("Editing Ticket: Status Code: {0}".format(r.status_code))
             logging.info("Edited ticket with the mentioned fields {0} - {1}".format(self.ticket_id, self.ticket_url))
+            return r
```
Comment (dmranck) [inline ticketutil/bugzilla.py]: Remove this for now. I'll be working on what to return in a future PR.

File: ticketutil/bugzilla.py [hunk 7]
```diff
@@ -350,8 +377,6 @@ def main():
     main() function, not directly callable.
     :return:
     """
-    print("Not directly executable")
```
Comment (dmranck) [inline ticketutil/bugzilla.py]: Keep these two lines in here.

File: ticketutil/bugzilla.py [hunk 8]
```diff
@@ -138,10 +180,7 @@ def _create_ticket_request(self, params):
         """
         # Attempt to create ticket.
         try:
-            if self.token:
```
Comment (dmranck) [inline ticketutil/bugzilla.py]: Do you not need to add {'token': self.token} to each request call anymore for HTTP Basic?

Comment (K-Gazdikova) [inline ticketutil/bugzilla.py]: removed.

File: ticketutil/bugzilla.py (hunk 4 above)
Comment (K-Gazdikova) [inline ticketutil/bugzilla.py]: updated.

File: ticketutil/bugzilla.py (hunk 5 above)
Comment (K-Gazdikova) [inline ticketutil/bugzilla.py]: updated.

File: ticketutil/bugzilla.py (hunk 6 above)
Comment (K-Gazdikova) [inline ticketutil/bugzilla.py]: updated.

File: ticketutil/bugzilla.py (hunk 7 above)
Comment (K-Gazdikova) [inline ticketutil/bugzilla.py]: Added.

**Comment by dmranck** [comment]
Just functionally tested authenticating with api_key and it works well. This is ready to be merged once docs have been updated.

**Pull request merged**